Add guide on using files generated by GenAi-Perf

dyastremsky opened the pull request:
This pull request adds a document outlining how to use the files generated by GenAi-Perf. The aim of the guide is to be concise and clear, providing a basic starting point for reading and using these files.

dyastremsky commented:
@debermudez, do you know if profile_json.json and profile_export.json redundant? I don't see in the code were profile_json.json gets generated and used. I don't know when it started getting generated. They look similar but they are not the same.

dyastremsky commented on src/c++/perf_analyzer/genai-perf/docs/files.md:
```diff
@@ -0,0 +1,130 @@
+
+##### GZIP Files
+
+- all_data.gzip: Aggregated performance data from all collected metrics.
```
@debermudez Do you know if this is per request? It's also a bit confusing, as I'm seeing datapoint 0 have request_throughputs but no other rows have that. And I don't know how token-to-token times factor in for this chart. Any clarification here coming from you would be great. Feel free to even commit a description directly, if easier.

Also, did I capture per-request versus per-token correctly for the other GZIP files?

debermudez replied:
All of the data in a parquet file was a request from mark moyou.  
If the data is confusing however, we should clean it up. Some of the data only had 1 datapoint while the others have data for every request which led to the discrepancy.

dyastremsky replied:
There is an offline conversation with Mark and Elias. It sounds like the columns are all concatenated, so the values in the rows have no real relationship with one another. Depending on Mark's needs, we may be able to remove this file. If we do so in a separate PR, we can remove this line.

dyastremsky commented on src/c++/perf_analyzer/genai-perf/docs/files.md:
```diff
@@ -0,0 +1,130 @@
+##### JSON Files
+
+- llm_inputs.json: This contains the input prompts provided to the LLM during testing.
+- profile_export.json: This is provided by Perf Analyzer and contains the timestamps
```
@matthewkotila Any suggestions for improving this description? I don't know if we have documentation somewhere to point to. It sounds like we might want to even provide a tutorial for working with profile_export.json, though that'd likely live better in PA even if GenAi-Perf is the current use case.

matthewkotila replied:
There's unfortunately no documentation anywhere currently describing the profile export JSON. There is a schema file that technically defines it but it isn't really a tutorial. It would be good to add documentation to PA to describe this file. I think this short description is good enough for now.

dyastremsky replied:
Gotcha, thanks for clarifying. That sounds good!

debermudez commented:
> generated and used. I don't know when it started getting generated. They look similar but they are not the same.

profile_json.json is a typo i introduced. it is profile_export.json. It should be fixed in the main branch and the release.

debermudez commented on src/c++/perf_analyzer/genai-perf/docs/files.md:
```diff
@@ -0,0 +1,130 @@
+- input_tokens_vs_generated_tokens.gzip: This contains data on the number of
+input tokens versus the number of generated tokens for each request.
+- request_latency.gzip: This contains the latency for each request.
+- time_to_first_token.gzip: Thsi contains the time to first token for each request.
```
```suggestion
- time_to_first_token.gzip: This contains the time to first token for each request.
```

dyastremsky commented:
> > generated and used. I don't know when it started getting generated. They look similar but they are not the same.
> 
> profile_json.json is a typo i introduced. it is profile_export.json. It should be fixed in the main branch and the release.

Thanks, removed!

matthewkotila commented on src/c++/perf_analyzer/genai-perf/docs/files.md:
```diff
@@ -0,0 +1,130 @@
+- profile_export.json: This is provided by Perf Analyzer and contains the timestamps
+for each event in the lifecycle of each request. This is low-level data used to calculate
+metrics by GenAi-Perf.
+- profile_json.json: *Confirm with team. This looks similar to profile_export.json but a bit difference.*
```
What is this file?

dyastremsky replied:
I removed this line. A typo in a PR introduced it before, but it's since been removed.
